test/types/community/collection/recursive-types.test-d.ts

```diff
@@ -1,23 +1,70 @@
-import { expectError } from 'tsd';
+import { expectAssignable, expectError, expectNotAssignable, expectNotType } from 'tsd';
 
-import type { Collection } from '../../../../src';
+import type { Collection, Filter, UpdateFilter } from '../../../../src';
 
 /**
  * mutually recursive types are not supported and will not get type safety
  */
-interface A {
-  b: B;
+interface Author {
+  name: string;
+  favoritePublication: Book;
 }
 
-interface B {
-  a: A;
+interface Book {
+  title: string;
+  author: Author;
 }
 
-declare const mutuallyRecursive: Collection<A>;
-//@ts-expect-error
-mutuallyRecursive.find({});
-mutuallyRecursive.find({
-  b: {}
+expectAssignable<Filter<Author>>({
+  favoritePublication: {
+    title: 'book title',
+    author: {
+      name: 'author name'
+    }
+  }
+});
+expectNotType<UpdateFilter<Author>>({
```

Comment:
what's the purpose of this specific test?

Reply:
Added a comment, its just double checking that adding the `& Document` case doesn't lose checking when you aren't even using dot notation